### PR TITLE
[YUNIKORN-2192] Adjust the instruction of create/update ConfigMap

### PR DESCRIPTION
**Pull request by targetoee**
### What is this PR for?
There is a simpler way to create/update a ConfigMap (just use `kubectl apply -f` instead of using `kubectl create configmap` with arguments). 
This PR adjusts the instructions to make it easier for users.

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [x] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
N/A

### What is the Jira issue?
[YUNIKORN-2192](https://issues.apache.org/jira/projects/YUNIKORN/issues/YUNIKORN-2192)

### How should this be tested?
N/A

### Screenshots (if appropriate)
N/A

### Questions:
N/A

### CODE REVIEW

File: docs/developer_guide/deployment.md [hunk 1]
```diff
@@ -60,13 +60,13 @@ This kubernetes environment can be either local or remote.
...
 curl -o yunikorn-configs.yaml https://raw.githubusercontent.com/apache/yunikorn-k8shim/master/deployments/scheduler/yunikorn-configs.yaml
 ```
-- modify the content of yunikorn-configs.yaml file as needed, and create ConfigMap in kubernetes:
+- modify the content of yunikorn-configs.yaml file as needed, and apply yunikorn-configs.yaml file in kubernetes:
 ```
-kubectl create configmap yunikorn-configs --from-file=yunikorn-configs.yaml
+kubectl apply -f yunikorn-configs.yaml
 ```
-- Or update ConfigMap in kubernetes:
+- if configuration changes are needed, update the yunikorn-configs.yaml file and apply it again:
```
[the remark this answers is not in the document]
Comment (craigcondit) [reply docs/developer_guide/deployment.md:67]: I think we can replace this entire section with a link to the user guide configuration section. Manually creating a config has not been necessary to bootstrap YuniKorn for quite some time now, and should only be needed during development if testing some features which require non-default configuration.

[the remark this answers is not in the document]
Comment (targetoee) [reply docs/developer_guide/deployment.md]: I agree with @chia7712 that this part is redundant. The instruction already cover both creating and updating.
Also the opinion of @craigcondit is reasonable. I add a reference link to it. Thanks for the advice.

File: docs/developer_guide/deployment.md [hunk 2]
```diff
@@ -60,18 +60,15 @@ This kubernetes environment can be either local or remote.
...
-- Or update ConfigMap in kubernetes:
-```
-kubectl create configmap yunikorn-configs --from-file=yunikorn-configs.yaml -o yaml --dry-run=client | kubectl apply -f -
+kubectl apply -f yunikorn-configs.yaml
 ```
 - check if the ConfigMap was created/updated correctly:
 ```
 kubectl describe configmaps yunikorn-configs
 ```
+- for more configuration detail, see [Service Configuration](https://yunikorn.apache.org/docs/user_guide/service_config).
```
Comment (wilfred-s) [inline docs/developer_guide/deployment.md:71]: we should be using a relative link like `user_guide/service_config.md`

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: docs/developer_guide/deployment.md (hunk 2 above)
Comment (targetoee) [reply docs/developer_guide/deployment.md]: Done

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request closed without merging**